web/app/fr/gospeak/web/auth/services/AuthSrv.scala

```diff
@@ -110,18 +112,65 @@ class AuthSrv(userRepo: AuthUserRepo,
     } yield authUser
   }
 
+  def createOrEdit(socialProfile: CommonSocialProfile): IO[AuthUser] = {
+    def create(login: Login): IO[AuthUser] = {
+      val now = Instant.now()
+      for {
+        data <- SocialProfile.from(socialProfile).toUserData.toIO
+        u <- userRepo.create(data, now)
+        request <- userRequestRepo.createAccountValidationRequest(u.email, u.id, now)
+        _ <- userRequestRepo.validateAccount(request.id, u.email, now)
+        _ <- userRepo.createLoginRef(login, u.id)
+      } yield AuthUser(socialProfile.loginInfo, u, Seq())
+    }
+
+    def edit(user: User, login: Login): IO[AuthUser] = {
+      val now = Instant.now()
+      for {
+        groups <- groupRepo.list(user.id)
+        email <- EmailAddress.from(socialProfile.email.getOrElse(user.email.value)).toIO
+        u <- userRepo.edit(user.copy(
+          firstName = socialProfile.firstName.getOrElse(user.firstName),
+          lastName = socialProfile.lastName.getOrElse(user.lastName),
+          email = email),
+          now = now)
+      } yield AuthUser(socialProfile.loginInfo, u, groups)
+    }
+
+    val login = Login(ProviderId(socialProfile.loginInfo.providerID),
+      ProviderKey(socialProfile.loginInfo.providerKey))
+
+    for {
+      optUser <- userRepo.find(login)
+      authUser <- optUser.fold(create(login))(u => edit(u, login))
+    } yield authUser
+  }
+
+
+  def socialProviders: Seq[SocialProvider] = socialProviderRegistry.providers
+
+  def provider(providerID: String): Option[SocialProvider] = socialProviders.find(_.id == providerID)
```

Comment:
Maybe take a `ProviderId` as parameter and the same in controller using a PathBinder... can improve typesafety